### PR TITLE
[#2602] [JSDoc] Add missing documentation for botbuilder-dialogs-declarative files

### PR DESCRIPTION
**Pull request by denscollo**
Addresses #2602

## Description
This PR adds the missing documentation in [botbuilder-dialogs-declarative](https://github.com/microsoft/botbuilder-js/tree/master/libraries/botbuilder-dialogs-declarative) library.
Some documentation is ported and adapted from [botbuilder-dotnet](https://github.com/microsoft/botbuilder-dotnet)

## Specific Changes
- Added JSDoc comments in all public methods and classes.

## Testing
The following image shows the ESLint Report with no JSDoc errors.
image

### CODE REVIEW

File: libraries/botbuilder-dialogs-declarative/src/resources/resourceExplorer.ts [hunk 1]
```diff
@@ -210,9 +210,15 @@ export class ResourceExplorer {
         return result;
     }
 
+    /**
+     * @protected
+     * Handler for on changed events.
+     * @param event Resource Change Event.
```
Comment (chrimc62) [inline libraries/botbuilder-dialogs-declarative/src/resources/resourceExplorer.ts:216]: >Resource Change Event [](start = 20, length = 21)

Don't really need to capitalize all of the words in this.

File: libraries/botbuilder-dialogs-declarative/src/resources/resourceProvider.ts [hunk 2]
```diff
@@ -85,9 +85,15 @@ export abstract class ResourceProvider {
      */
     public abstract refresh(): void;
 
+    /**
+     * @protected
+     * Actions to perform when the current object is changed.
+     * @param event Resource Change Event.
```
Comment (chrimc62) [inline libraries/botbuilder-dialogs-declarative/src/resources/resourceProvider.ts:91]: >Resource Change Event. [](start = 20, length = 22)

Don't need to capitalize all.

**Review by chrimc62 — APPROVED** [review]
:shipit:

**Review by joshgummersall — CHANGES_REQUESTED** [review]
In an effort to improve generated docs, properly linking types with `[link text](xref:uid)` would be really helpful.

See more details [here](https://github.com/microsoft/botbuilder-js/pull/2838#discussion_r499724516) and [here](https://github.com/microsoft/botbuilder-js/pull/2838#discussion_r499748705).

**Review by joshgummersall — APPROVED** [review]

**Pull request merged**